### PR TITLE
[Updated] MouseCursor

### PR DESCRIPTION
**Pull request by Tritaris**
Fixed scaling issue, moved some code to a separate file and refactoring.

### CODE REVIEW

File: Assets/Scripts/UI/MouseCursor.cs [hunk 1]
```diff
@@ -87,217 +71,114 @@ public void Update()
...
 
         Tile t = WorldController.Instance.GetTileAtWorldCoord(mc.GetMousePosition());        
         
-        // Placing furniture object
-        if (bmc.buildMode == BuildMode.FURNITURE)
-        {            
-            lowerRightString = World.current.furniturePrototypes[bmc.buildModeObjectType].Name;
+        if (mc.GetCurrentMode() == MouseController.MouseMode.BUILD)
+        {
+            // Placing furniture object
```
Comment (Randshot) [inline Assets/Scripts/UI/MouseCursor.cs]: Maybe add a period after the comment.
The same for the next one.

**Comment by RodRitter** [comment]
Something is wrong here. When I launch the game, I get 2 fps and stuck in cursor mode. Please fix this

**Comment by Randshot** [comment]
@Tritaris Is this relevant?
```
Invalid texture used for cursor - check importer settings
UnityEngine.Cursor:SetCursor(Texture2D, Vector2, CursorMode)
MouseCursor:BuildCursor() (at Assets/Scripts/UI/MouseCursor.cs:113)
MouseCursor:.ctor(MouseController, BuildModeController) (at Assets/Scripts/UI/MouseCursor.cs:54)
MouseController:.ctor(BuildModeController, FurnitureSpriteController, GameObject) (at Assets/Scripts/Controllers/MouseController.cs:55)
WorldController:Start() (at Assets/Scripts/Controllers/WorldController.cs:110)
```

**Comment by RodRitter** [comment]
Nevermind, it's great. Merged.

**Pull request merged**